lib/utils/analytics/index.js

```diff
@@ -115,34 +115,29 @@ async function report(payload, options = {}) {
         return null;
       }
     })(),
-    request(payload, { id }),
+    request([payload], { ids: [id] }),
   ]);
   return requestResult;
 }
 
 async function sendPending(options = {}) {
   const isForced = options && options.isForced;
   serverlessRunEndTime = null; // Needed for testing
-  if (options.serverlessExecutionSpan) {
-    options.serverlessExecutionSpan.then(markServerlessRunEnd, markServerlessRunEnd);
-  }
-  if (areAnalyticsDisabled && !isForced) return;
-  if (!cacheDirPath) return;
-  const limit = pLimit(3);
+  if (areAnalyticsDisabled && !isForced) return null;
+  if (!cacheDirPath) return null;
   let dirFilenames;
   try {
     dirFilenames = await fse.readdir(cacheDirPath);
   } catch (readdirError) {
     if (readdirError.code !== 'ENOENT') logError('Cannot access cache dir', readdirError);
-    return;
+    return null;
   }
 
-  if (!options.serverlessExecutionSpan) process.nextTick(markServerlessRunEnd);
+  process.nextTick(markServerlessRunEnd);
```

Comment:
`markServerlessRunEnd` was created to mark point, that allows to detect whether stats requests exceeded the command processing time. If it exceeds the debug log is shown

It hardly makes sense without `serverlessExecutionSpan`, and in normal  usage `serverlessExecutionSpan` is always passed to this function (at this point it's not clear for me why this conditional was added)

If we remove handling of `serverlessExecutionSpan`, then I believe we should also remove handling of `markServerlessRunEnd` and this debug log logic.

Still, now (as I reminded myself about that useful log) I think that probably it's good to leave support for `options.serverlessExecutionSpan` for the sake of this debug log working. WDYT?

I personnally have `SLS_STATS_DEBUG` set in my `~/.profile` to always have it displayed

Reply:
I guess it still might be useful - I've brought back the conditional that checks for `serverlessExecutionSpan` and removed marking it if it's not passed. We can also make it non-optional because as you mentioned in "normal" flow it's always passed (currently it's not passed in tests and I believe that was the reason to have this conditional). What do you think?